push-sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/apns/ApnsUtil.java

```diff
@@ -66,4 +73,32 @@ public static String readDefaultTopic(final byte[] keystore, final char[] passwo
 
         return null; // if no topic was found we try with null
     }
+
+    public static boolean checkValidity(final byte[] keystore, final char[] password) {
+        try {
+            final KeyStore keyStore = KeyStore.getInstance(KEYSTORE_TYPE);
+            keyStore.load(new ByteArrayInputStream(keystore), password);
+
+            final Enumeration<String> aliases = keyStore.aliases();
+
+            while (aliases.hasMoreElements()) {
+                final String alias = aliases.nextElement();
+                final X509Certificate certificate = (X509Certificate) keyStore.getCertificate(alias);
+
+
+                try{
+                    certificate.checkValidity();
+                } catch (CertificateExpiredException | CertificateNotYetValidException e) {
+                    LOGGER.error("Provided APNs .p12 file is not valid");
```

Comment:
with a lot of pushes, e.g. from hooks / automated schedulers etc, this might generate a lot of ERROR loggings...